tools/generate_features.py

```diff
@@ -88,174 +86,142 @@ def drop_close_bright_stars(
-                            }
-                        },
-                        "filter": {},
+    query = {
+        "query_type": "cone_search",
+        "query": {
+            "object_coordinates": {
+                "radec": {'quad': [ctr_ra, ctr_dec]},
+                "cone_search_radius": max_cone_radius,
+                "cone_search_unit": 'arcsec',
+            },
+            "catalogs": {
+                catalog: {
+                    # Select sources brighter than G magnitude 13:
+                    # -Conversion to Tycho mags only good for G < 13
+                    # -Need for exclusion radius only for stars with B <~ 13
+                    # -For most stars, if G > 13, B > 13
+                    "filter": {"phot_g_mean_mag": {"$lt": 13.0}},
+                    "projection": {
+                        "phot_g_mean_mag": 1,
+                        "bp_rp": 1,
+                        "coordinates.radec_geojson.coordinates": 1,
                     },
                 }
-                for dct in radec_split_list
-            ]
+            },
+            "filter": {},
+        },
+    }
 
-            q = kowalski_instance.batch_query(queries, n_treads=8)
-            for batch_result in q:
-                gaia_results = batch_result['data'][catalog]
-                gaia_results_dct.update(gaia_results)
-        else:
-            # Get Gaia EDR3 ID, G mag, BP-RP, and coordinates
-            query = {
-                "query_type": "cone_search",
-                "query": {
-                    "object_coordinates": {
-                        # "radec": dict(zip(id_slice, radec_geojson.transpose().tolist())),
-                        "radec": radec_dict,
-                        "cone_search_radius": query_radius_arcsec,
-                        "cone_search_unit": 'arcsec',
-                    },
-                    "catalogs": {
-                        catalog: {
-                            # Select sources brighter than G magnitude 13:
-                            # -Conversion to Tycho mags only good for G < 13
-                            # -Need for exclusion radius only for stars with B <~ 13
-                            # -For most stars, if G > 13, B > 13
-                            "filter": {"phot_g_mean_mag": {"$lt": 13.0}},
-                            "projection": {
-                                "phot_g_mean_mag": 1,
-                                "bp_rp": 1,
-                                "coordinates.radec_geojson.coordinates": 1,
-                            },
-                        }
-                    },
-                    "filter": {},
-                },
-            }
+    q = kowalski_instance.query(query)
 
-            q = kowalski_instance.query(query)
-            gaia_results = q['data'][catalog]
-            gaia_results_dct.update(gaia_results)
+    gaia_results = q.get('data')
+    gaia_results_dct.update(gaia_results['Gaia_EDR3'])
```

Comment:
Maybe make the chosen data release a variable?

Reply:
Agreed!